Sources/Proxy/Core/View2DProxy/index.js

```diff
@@ -93,6 +127,89 @@ function vtkView2DProxy(publicAPI, model) {
+   * Exact fitting requires useParallelRendering, and an active camera
+   */
+  const superLightResetCamera = publicAPI.lightResetCamera;
+  publicAPI.lightResetCamera = (bounds) => {
+    // Always reset camera first to set physicalScale, physicalTranslation and trigger events
+    const initialReset = superLightResetCamera(bounds);
+    if (
+      !model.cameraResetIsFitting ||
+      !model.useParallelRendering ||
+      !initialReset
+    ) {
+      return initialReset;
+    }
+
+    // For each visible prop get the smallest possible convex hull using bounds corners
+    const visiblePoints = [];
+    if (bounds) {
+      // Bounds are given as argument, use their corners
+      vtkBoundingBox.getCorners(bounds, visiblePoints);
+    } else {
+      publicAPI.getRepresentations().forEach((reprenstationProxy) =>
+        reprenstationProxy
+          .getActors()
+          .concat(reprenstationProxy.getVolumes())
+          .forEach((prop) => visiblePoints.push(...getPropCoarseHull(prop)))
+      );
+    }
+
+    // Get the bounds in view coordinates
+    const viewBounds = vtkBoundingBox.reset([]);
+    const viewMatrix = model.camera.getViewMatrix();
+    mat4.transpose(viewMatrix, viewMatrix);
+
+    for (let i = 0; i < visiblePoints.length; ++i) {
+      const point = visiblePoints[i];
+      vec3.transformMat4(point, point, viewMatrix);
+      vtkBoundingBox.addPoint(viewBounds, ...point);
+    }
+
+    // Compute focal point and position
+    const viewFocalPoint = vtkBoundingBox.getCenter(viewBounds);
+    // Camera position in view coordinates is the center of the bounds in XY
+    // and the maximum bound + 1 in Z
+    const viewPosition = [
+      viewFocalPoint[0],
+      viewFocalPoint[1],
+      viewBounds[5] + 1,
+    ];
+    const inverseViewMatrix = new Float64Array(16);
+    const worldFocalPoint = new Float64Array(3);
+    const worldPosition = new Float64Array(3);
+    mat4.invert(inverseViewMatrix, viewMatrix);
+    vec3.transformMat4(worldFocalPoint, viewFocalPoint, inverseViewMatrix);
+    vec3.transformMat4(worldPosition, viewPosition, inverseViewMatrix);
+
+    // Compute parallel scale
+    const view = model.renderer.getRenderWindow().getViews()[0];
+    const dims = view.getViewportSize(model.renderer);
+    const aspect = dims[0] / dims[1];
+    const xLength = vtkBoundingBox.getLength(viewBounds, 0);
+    const yLength = vtkBoundingBox.getLength(viewBounds, 1);
+    const parallelScale = 0.5 * Math.max(yLength, xLength / aspect);
+
+    if (parallelScale <= 0) {
+      return initialReset;
+    }
+
+    // Compute bounds in world coordinates
+    const worldBounds = vtkBoundingBox.transformBounds(
+      viewBounds,
+      inverseViewMatrix
+    );
+
+    model.camera.setFocalPoint(...worldFocalPoint);
+    model.camera.setPosition(...worldPosition);
+    model.camera.setParallelScale(parallelScale);
+    model.renderer.resetCameraClippingRange(worldBounds);
```

Comment:
it looks like this function can split in 2 : one part to compute the parameters, the other part to apply the parameters.